spinalcordtoolbox/scripts/sct_extract_metric.py

```diff
@@ -359,12 +360,7 @@ def main(fname_data, path_label, method, slices, levels, fname_output, labels_us
     if vertebral_levels:
         im_vertebral_labeling = Image(fname_vertebral_labeling).change_orientation("RPI")
 
-        # Get dimensions of input Image and the vertebral labeling file
-        nx, ny, nz = input_im.data.shape
-        nx_vertebral, ny_vertebral, nz_vertebral = im_vertebral_labeling.data.shape
-
-        # Check dimensions consistency between the input Image and the vertebral labeling file
-        if (nx, ny, nz) != (nx_vertebral, ny_vertebral, nz_vertebral):
+        if check_dimensions_match(input_im=input_im, im_vertebral_labeling=im_vertebral_labeling):
             printv('\nERROR: Metric data and vertebral labeling file DO NOT HAVE SAME DIMENSIONS.', 1, type='error')
```

Comment:
Is there a reason this isn't an exception?

```suggestion
        if input_im.shape != im_vertebral_labeling.shape:
            raise ValueError('\nERROR: Metric data and vertebral labeling file DO NOT HAVE SAME DIMENSIONS.')
```

Reply:
The file that you are referring to is outdated, I do the `check_dimensions_match` in `aggregate_per_slice_or_level()`. Also, catch the `ValueError` here: https://github.com/neuropoly/spinalcordtoolbox/blob/bfc109083aec1536dbbac4c7c77cb752bd10ee3f/spinalcordtoolbox/aggregate_slicewise.py#L252-L255